src/annotations/modifiers/AnnotationItemModifier.cpp

```diff
@@ -93,6 +100,14 @@ void AnnotationItemModifier::handleMouseRelease(QList<AbstractAnnotationItem *>
 	handleSelection();
 }
 
+void AnnotationItemModifier::handleMouseDoubleClick()
+{
+	if (mItemSelector->selectedItems().size() == 1) {
+		mIgnoreNextReleaseEvent = true;
```

Comment:
As specified in the docs, after the double-click event, the scene receives a release event. Handling the release event will clear the selection (and the item properties in the left sidepanel); reason for which we want to ignore this release event.

Reply:
I don't understand why a a release would clear the selection? I just removed this bool variable from the code and the double-click edit seems to be working the same. As far as I remember, when you have the select tool activated and click an item (click followed by release), the item gets selected, that's what I see when I try out the annotator.

Reply:
I can merge like without the variable if you want, you don't need to append to the PR.

Reply:
```cpp
AnnotationItemModifier::handleMouseDoubleClick
    emit itemEdit()
        AnnotationArea::enableEditing
            mItemModifier->clear();
                mItemSelector->clearSelection();
// no item is selected at this point

AnnotationItemModifier::handleMouseRelease
    handleSelection();
        if (count == 0) {
		clear();
		emit itemsDeselected();
	}
            mSettingsProvider->activateSelectTool();

// item is now in edit mode, but, because of the call to activateSelectTool, the properties panel is cleared, we don't want this
```

Reply:
But how does that differ from the current behvior? When I right click on a text item and select "edit" the tool jumps also back to to select tool. Looks kind of same to me. 

What I'm trying to avoid is tracking click state in the modifier. Can we introduce a new class that is used in the modifier that takes care of making item editable? Something like AnnotationItemEditor that gets position of the double click and all items from the scene and then searches for any item that can be edited at that position. If it finds one, it makes it editable and remembers that there is an item that is currently being edited. You can ask the AnnotationItemEditor if he knows about any items that is being edited and based on that information you could act in the modifier (clearing selection, emit signals, what ever).

Reply:
@DamirPorobic 
When you select 'Edit' from the context menu it jumps to the select tool, but it does not clear the properties panel.
If we remove the `mIgnoreNextReleaseEvent`, it will jump to the select tool and also clear the properties panel.

Reply:
Exactly, that's the behavior that I saw but kind of looks correcter to me to clear the properties panel when the select tool is select, you can't change anything with those properties. 

I'll have a look at the your last commits.

Reply:
> Exactly, that's the behavior that I saw but kind of looks correcter to me to clear the properties panel when the select tool is select, you can't change anything with those properties.
> 
> I'll have a look at the your last commits.

Ah, understood.